### ISSUE TITLE
File Size (in brackets) added to all document links

### ISSUE THREAD

**Issue by AWearring** [localgovdrupal/localgov_base#363]
File Size (in brackets) added to all document links after Base Theme change from "stable" to "stable9"
https://www.evernote.com/shard/s369/sh/b993720f-afb2-8e06-8b9e-9cc57a3cdd79/e4028302011765b4e4022233bbd75531

**Comment by finnlewis** [comment]
thanks @AWearring !

Looks to be coming from https://git.drupalcode.org/project/drupal/-/blob/9.5.2/core/themes/stable9/templates/field/file-link.html.twig

Maybe we can just override it in https://github.com/localgovdrupal/localgov_base/tree/1.x/templates/field

**Comment by finnlewis** [comment]
Confirmed locally with a default install and localgov_demo module enabled.

At https://localgov.ddev.site/test-subsite-demo-all-components/test-pages-documents I see: 

![image](https://user-images.githubusercontent.com/326588/215813437-608d1c68-a25d-4f43-8ff7-9c9ba758f1ad.png)

If I then override the template in the localgov_scarfolk theme with the template from stable theme (
https://git.drupalcode.org/project/drupal/-/blob/9.5.2/core/themes/stable/templates/field/file-link.html.twig)

We now see: 

![image](https://user-images.githubusercontent.com/326588/215813697-01ca04dd-64d3-4884-916f-409ff4f4989b.png)

**Comment by markconroy** [comment]
Maybe we could leave this as is, and remove the custom code we have that creates the file type/file size addition. One less thing for us to maintain?

**Comment by finnlewis** [comment]
Sounds tempting, thanks @markconroy, where is the custom code? 

One thing that does look like it would change is currently the whole label is a link, including the file size. By default in stable9 the links come first and the filesize is not linked. 

Whether one is preferable or not is a question, but this would change the behaviour for people downstream.

Any thoughts or preference @AWearring ?

**Comment by markconroy** [comment]
I'm not sure exactly where the custom code is, but I _think_ it might be here:
https://github.com/localgovdrupal/localgov_paragraphs/blob/2.x/modules/localgov_subsites_paragraphs/localgov_subsites_paragraphs.module#L192

and here:
https://github.com/localgovdrupal/localgov_paragraphs/blob/2.x/modules/localgov_subsites_paragraphs/templates/paragraph--localgov-documents.html.twig

**Comment by AWearring** [comment]
A couple of thoughts:

- From an Accessibility perspective do we *need* to have the file type and size included in the link text? 
- Could this information be visually hidden instead
- For localgov_documents I'd prefer the file type and size to be in separate columns


I think the custom code is in localgov_base.theme (https://github.com/localgovdrupal/localgov_base/blob/1.x/localgov_base.theme) - lines 82 - 104  [Edited] **Confirmed** removing these lines removes the file type and size

**Comment by markconroy** [comment]
@AWearring Good find. Though I'm sorry that's in the theme file, it's not something I'm happy with there. If it's in a template file, you can more easily override the template.

Looks like we copied this from the `localgov_theme` theme when we created the `localgov_base` theme.

**Comment by andybroomfield** [comment]
There will be implications if we change the code in the module for anything that is relying on the way its printed, or the extra variables it generates. Thinking any theme that makes use of that code. If this is going to be a change it should be in the theme.

It looks like [this is part of the Croydon design system](https://www.croydon.gov.uk/design-and-content-guidelines/components/download-list) and may have origin at BHCC, and ultimatly GDS.

It is considered an accessbility requirement from our side to show the file size and the file type, so we need to be able to keep them visible.

**Comment by andybroomfield** [comment]
When BHCC had to override this from stable9, we did the following:
````php
/**
  * Implements hook_preprocess_HOOK().
  */
 function bhcc_base_preprocess_file_link(&$variables) {

   // Don't show the file size.
   $variables['file_size'] = FALSE;
 }
````
and lets us keep the existing behaviour.
Maybe we then look at changing the pattern, though that is something I think would be equivalent to a breaking api change.

**Comment by markconroy** [comment]
I guess @finnlewis's approach is a good fix for now, and we can look at updating/removing the custom code later. It's not very complex code so is fairly easy to maintain, although it has created issues and 2 PRs against it already for messing up things a little when anonymous users are uploading files through webform.

- #273 
- #293

**Comment by andybroomfield** [comment]
I'm not finding the pattern in GDS, though there is this discussion [GDS link discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/20).
I see @finnlewis has a PR open, and if that resolves the issue I would go with that for now.

**Comment by AWearring** [comment]
Having spoken to the Big Blue Door dev team we would prefer the BHCC override solution instead of using an additional twig file

**Comment by AWearring** [comment]
I've just tested the BHCC solution and it removes both the File Size added by stable9 and the File Size added in localgov_base.theme. So it would seem that the solution may have to be the one @finnlewis suggested (https://github.com/localgovdrupal/localgov_base/pull/364)

**Comment by finnlewis** [comment]
Thanks @AWearring - I'm also keen to hear from the Big Blue Door dev team on preferences.

Currently I think the pull request essentially re-instates the behaviour that was there before, by effectively reverting the template change at the localgov_base level.

**Comment by AWearring** [comment]
@finnlewis I've just tried something different and it now works
Initially I added `$variables['file_size'] = FALSE;` to an existing `preprocess_file_link` function, I've created a completely new function and it behaves the way we want it to.

**Comment by finnlewis** [comment]
@markconroy @AWearring should we go ahead with the proposed fix to reinstate the original template https://github.com/localgovdrupal/localgov_base/pull/364 and consider changes on a different pull request?

**Comment by AWearring** [comment]
@finnlewis @markconroy We've implemented the BHCC code @andybroomfield suggested at West Lindsey (also for Rutland and North Kesteven) - I've tested the fix (#364) with the BHCC code in place and there are no issues, so I'd be happy to proceed with the proposal.

**Comment by willguv** [comment]
@finnlewis @markconroy @stephen-cox do you know if this fix has been merged? Cheers

**Comment by stephen-cox** [comment]
@willguv This looks have have been fixed with the 1.3.1 release of the theme: https://github.com/localgovdrupal/localgov_base/releases/tag/1.3.1
